include/alloca.h

```diff
@@ -37,6 +37,11 @@
 #  undef  __alloca
 #  define alloca(size)   _alloca(size)
 #  define __alloca(size) _alloca(size)
+#elif defined(__TASKING__)
+#  undef  alloca
+#  undef  __alloca
+#  define alloca(size)   malloc(size)
```

Comment:
memory leak?

Reply:
Yes, seems still need to improve compatibility on the application side

Reply:
https://github.com/apache/nuttx-apps/pull/2291

Reply:
> can we change sp manually?

how to restore the sp to original position ? setjmp/longjmp?

Reply:
sp should be store in stack in this case

Reply:
if alloca can't implement easily with tasking compiler, it's better to keep alloca is undefined to force the user modify the code to not use it.